### PR TITLE
Contrastive Loss implementation

### PR DESCRIPTION
**Pull request by shardulparab97**
@zoq @rcurtin  I have added the implementation of Contrastive Loss. Can you please let me know your thoughts about the same. :smile: 

I will be adding the tests today itself.

### CODE REVIEW

**Comment by shardulparab97** [comment]
#1699

**Comment by shardulparab97** [comment]
Sorry for the late update - was caught up with college exams.
I am unable to understand the error given out by Jenkins:

**CMake Error at CMakeLists.txt:42 (add_executable):
  CXX_STANDARD is set to invalid value '17'**

**Comment by zoq** [comment]
Sorry for the confusion, that's internal build issue.

**Comment by zoq** [comment]
Do you like to fix the merge issue or should I do it, just let me know.

**Comment by shardulparab97** [comment]
Thanks @zoq! Can you please let me know about solving the merge issue.

**Comment by zoq** [comment]
See https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request I guess you can also use the github web editor right inside the PR.

**Comment by zoq** [comment]
@mlpack-jenkins test this please

**Comment by zoq** [comment]
Any idea why the test might fail?

**Comment by shardulparab97** [comment]
@zoq this is a silly question, but I wanted to ask whether all the warnings for Style checks also need to resolved to get a pass for Style Checks

**Comment by zoq** [comment]
If we can solve the style issues that would be great, it looks like almost all of the issues are trailing spaces. Most editors do provide an option to remove them automatically.

File: src/mlpack/tests/loss_functions_test.cpp [hunk 1]
```diff
@@ -356,44 +357,58 @@ BOOST_AUTO_TEST_CASE(GradientReconstructionLossTest)
 }
 
 /*
- * Simple test for the dice loss function.
```
Comment (manish7294) [inline src/mlpack/tests/loss_functions_test.cpp]: Hi @shardulparab97, I see you have removed dice loss function. I guess either it is done mistakenly or is there any particular reason for removing it?

Comment (shardulparab97) [reply src/mlpack/tests/loss_functions_test.cpp]: Hi @manish7294, when I was working on this PR, the dice loss was not yet merged into the master.
I used the code in that PR as a reference, so hence it has come up like this :)

Comment (manish7294) [reply src/mlpack/tests/loss_functions_test.cpp]: That explains it. Maybe you could make the required change here :)

**Pull request closed without merging**